pages/index.js

```diff
@@ -1,87 +1,114 @@
-import Link from "next/link";
-import { Fragment } from "react";
-import Default from "../components/layout/default";
+import { useEffect, useState } from "react"
+import Link from "next/link"
+import Default from "../components/layout/default"
+import { get } from "../lib/api"
+import { normalizeAnchor } from "../lib/utils"
 
-const projectsData = [
-  {
-    id: 0,
-    label: "Pixel Retail",
-    slug: "/pixel-retail",
-  },
-  {
-    id: 1,
-    label: "Dating App",
-    slug: "/dating-app",
-  },
-  {
-    id: 2,
-    label: "INVESTMENT WEBSITE",
-    slug: "/dating-app",
-  },
-  {
-    id: 3,
-    label: "CHAMPAGNE WEBSITE",
-    slug: "/dating-app",
-  },
-];
-const otherProjectsData = [
-  {
-    id: 0,
-    label: "Pixel Retail",
-    slug: "/pixel-retail",
-  },
-  {
-    id: 1,
-    label: "Dating App",
-    slug: "/dating-app",
-  },
-  {
-    id: 2,
-    label: "INVESTMENT WEBSITE",
-    slug: "/dating-app",
-  },
-  {
-    id: 3,
-    label: "CHAMPAGNE WEBSITE",
-    slug: "/dating-app",
-  },
-];
+export async function getServerSideProps() {
+  const { data, message, isError } = await get("http://localhost:3000/project")
+  return {
+    props: {
+      projects: data,
+      message,
+      isError,
+    },
+  }
+}
+
+function Homepage({ projects, message, isError }) {
+  const mainProjects = projects.filter(
+    (project) => project.main_project === true
+  )
+  const otherProjects = projects.filter(
+    (project) => project.main_project === false
+  )
```

Comment:
Let's do some optimization here 
- wrap computations in `useMemo`
- loop over the array only once and either push in `mainProjects` array, either in `otherProjects`, then return both arrays from `useMemo`